src/async-script-loader.js

```diff
@@ -1,197 +1,211 @@
-import React from "react";
+import { Component, createElement } from "react";
 import PropTypes from "prop-types";
 
 let SCRIPT_MAP = {};
 
 // A counter used to generate a unique id for each component that uses the function
 let idCount = 0;
 
-export default function makeAsyncScript(Component, getScriptURL, options) {
+export default function makeAsyncScript(getScriptURL, options) {
```

Comment:
The below diff looks intense, but mostly just indentation due to HOC 😢 

biggest changes are:
- the added `(props, context)` to the `constructor`
- added comments to steps inside of `componentDidMount`
- passing the `entry` object to `asyncScriptOnLoad`

Reply:
can remove the whitespace from the diff => https://github.com/dozoisch/react-async-script/pull/34/files?w=1

Reply:
^ that's awesome... i need to remember that for all time.. lol